### PR TITLE
Fixed DB2 Profiler

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Fix https://github.com/open-metadata/OpenMetadata/issues/6967
Fix https://github.com/open-metadata/OpenMetadata/issues/7625
Fix https://github.com/open-metadata/OpenMetadata/issues/6953

#
### Type of change :

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/orm_profiler/metrics/static/column_count.py [hunk 1]
```diff
@@ -15,9 +15,32 @@
...
+
+
+@compiles(ColunCountFn, Dialects.IbmDbSa)
+@compiles(ColunCountFn, Dialects.Db2)
+def _(element, compiler, **kw):
+    """Returns column count for db2 database and handles casting variables.
+    If casting is not provided for variables, db2 throws error.
+    """
+    proc = compiler.process(element.clauses, **kw)
+    return f"CAST({proc} AS BIGINT)"
```
Comment (TeddyCr) [inline ingestion/src/metadata/orm_profiler/metrics/static/column_count.py:43]: Is this to handle the type not matching when it is None?

Comment (OnkarVO7) [reply ingestion/src/metadata/orm_profiler/metrics/static/column_count.py:43]: @TeddyCr it's not for `None`.

The query in db2 does not fail for `NULL` datatypes as we were suspecting earlier.
The query fails when we have variables in the query i.e the `?` thing. If this is not type casted into a DB2 datatype the query fails for DB2
The logic written here and other places is to handle this use case

**Review by TeddyCr — APPROVED** [review]
In `ingestion/src/metadata/orm_profiler/orm/functions/sum.py` is it expected for us to CAST twice?

File: ingestion/src/metadata/orm_profiler/orm/functions/sum.py [hunk 2]
```diff
@@ -58,3 +58,11 @@ def _(element, compiler, **kw):
...
     proc = compiler.process(element.clauses, **kw)
     return f"SUM({proc})"
+
+
+@compiles(SumFn, Dialects.IbmDbSa)
+@compiles(SumFn, Dialects.Db2)
+def _(element, compiler, **kw):
+    """Handle the case for DB2 where it requires to type cast the variables"""
+    proc = compiler.process(element.clauses, **kw).replace("?", "CAST(? AS INT)")
+    return f"SUM(CAST({proc} AS BIGINT))"
```
[the remark this answers is not in the document]
Comment (OnkarVO7) [reply ingestion/src/metadata/orm_profiler/orm/functions/sum.py:68]: It is expected since the `?` requires it's own casting

**Pull request merged**